recipe/bld.bat

```diff
@@ -1,24 +1,87 @@
+@echo on
+
+:: This step is required when building from raw source archive
+:: make generate --jobs %CPU_COUNT%
+:: if errorlevel 1 exit /b 1
+
+:: Duplicate lists because of https://bitbucket.org/icl/magma/pull-requests/32
+set "CUDA_ARCH_LIST=sm_50,sm_60,sm_70,sm_80"
+set "CUDAARCHS=50-virtual;60-virtual;70-virtual;80-virtual"
+
+if "%cuda_compiler_version%"=="11.8" (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST%,sm_90"
+  set "CUDAARCHS=%CUDAARCHS%;90-virtual"
+
+) else if "%cuda_compiler_version:~0,3%"=="12." (
+  set "CUDA_ARCH_LIST=%CUDA_ARCH_LIST%,sm_90"
+  set "CUDAARCHS=%CUDAARCHS%;90-virtual"
+
+) else (
+  echo Unsupported CUDA version. Please update build.bat
+  exit /b 1
+)
+
 md build
+cd build
 if errorlevel 1 exit /b 1
 
-cmake ^
-  -S . ^
-  -B build ^
+:: Set MKLROOT and CMAKE_PREFIX_PATH for MKL detection
+set "MKLROOT=%LIBRARY_PREFIX%"
+set "CMAKE_PREFIX_PATH=%LIBRARY_PREFIX%;%CMAKE_PREFIX_PATH%"
+
+:: Set BLA_VENDOR environment variable so CMake can find MKL
+set "BLA_VENDOR=Intel10_64lp"
+
+:: Explicitly set MKL library using the single-dynamic runtime (mkl_rt.lib)
+:: This is what's provided by mkl-devel on Windows defaults channel
+set "LAPACK_LIBRARIES=%LIBRARY_PREFIX%\lib\mkl_rt.lib"
+
+:: Must add --use-local-env to NVCC_FLAGS otherwise NVCC autoconfigs the host
+:: compiler to cl.exe instead of the full path. MSVC does not accept a
+:: C++11 standard argument, and defaults to C++14
+:: https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-160
+:: https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-160
+cmake %SRC_DIR% ^
   -G "Ninja" ^
-  -DUSE_FORTRAN=OFF ^
-  -DMAGMA_ENABLE_CUDA=ON ^
+  -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS:BOOL=ON ^
   -DCMAKE_BUILD_TYPE=Release ^
-  -DGPU_TARGET="Fermi Kepler Maxwell Pascal Volta Turing Ampere" ^
-  -DMAGMA_WITH_MKL ^
-  -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
-  -DCUDA_NVCC_FLAGS="--use-local-env" ^
+  -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
+  -DGPU_TARGET="%CUDA_ARCH_LIST%" ^
+  -DMAGMA_ENABLE_CUDA:BOOL=ON ^
+  -DUSE_FORTRAN:BOOL=OFF ^
+  -DMAGMA_WITH_MKL:BOOL=ON ^
+  -DMKLROOT=%LIBRARY_PREFIX% ^
+  -DBLA_VENDOR=Intel10_64lp ^
+  -DLAPACK_LIBRARIES="%LAPACK_LIBRARIES%" ^
+  -DCMAKE_CUDA_FLAGS="--use-local-env -Xfatbin -compress-all -Wno-deprecated-gpu-targets" ^
+  -DCMAKE_CUDA_SEPARABLE_COMPILATION:BOOL=OFF
 if errorlevel 1 exit /b 1
 
-cmake --build build ^
-      --config Release ^
-      --parallel %CPU_COUNT% ^
-      --target magma magma_sparse ^
-      --verbose
+cmake --build . ^
+    --config Release ^
+    --parallel %CPU_COUNT% ^
+    --target magma magma_sparse ^
+    --verbose
 if errorlevel 1 exit /b 1
 
-cmake --install .
```

Comment:
For my curiosity, why do we prefer to manually copy the files instead of setting CMAKE_INSTALL_PREFIX and using cmake --install?